pkg/controller/vsphere/reconciler.go

```diff
@@ -1088,6 +1100,66 @@ func getDiskSpec(s *machineScope, devices object.VirtualDeviceList) (types.BaseV
 	}, nil
 }
 
+func getAdditionalDiskSpecs(s *machineScope, devices object.VirtualDeviceList, datastore *object.Datastore) ([]types.BaseVirtualDeviceConfigSpec, error) {
+	var diskSpecs []types.BaseVirtualDeviceConfigSpec
+
+	klog.InfoS("About to iterate through disks", "disk", s.providerSpec.Disks)
+
+	klog.Infof("Feature Gates: %v", s.featureGates)
+	// Only add additional disks if the feature gate is enabled.
+	if len(s.providerSpec.Disks) > 0 && !s.featureGates.Enabled(featuregate.Feature(apifeatures.FeatureGateVSphereMultiDisk)) {
+		return nil, machinecontroller.InvalidMachineConfiguration(
+			"machines cannot contain additional disks due to VSphereMultiDisk feature gate being disabled")
+	}
```

Comment:
This needs to happen at the providerSpec validation webhook level as well

Reply:
Ok, I'll copy this up this up there too.  Thanks!